test/e2e/operator_test.go

```diff
@@ -152,6 +158,89 @@ func TestOperatorAvailable(t *testing.T) {
 	}
 }
 
+func TestExternalDNSWithRoute(t *testing.T) {
+	t.Log("Ensuring test namespace")
+	err := kubeClient.Create(context.TODO(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNamespace}})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("Failed to ensure namespace %s: %v", testNamespace, err)
+	}
+
+	t.Log("Creating external dns instance with source type route")
+	extDNS := helper.buildOpenShiftExternalDNS(testExtDNSName, hostedZoneID, hostedZoneDomain)
+	if err := kubeClient.Create(context.TODO(), &extDNS); err != nil {
+		t.Fatalf("Failed to create external DNS %q: %v", testExtDNSName, err)
+	}
+	defer kubeClient.Delete(context.TODO(), &extDNS)
+
+	// create a service of type ClusterIP without annotation
+	t.Log("Creating test route service")
```

Comment:
Do you really need the service? When I tried the route creation I put a service name which didn't exist and OCP admitted it. We'll not have the router configured properly without the service, so no e2e connectivity but we don't test this part, only the resolution of the router's IP.

Reply:
ok, it was failing on the schema registration, i thought we need the service, will remove it now 👍🏽